src/modules/profile/profile.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { APIResponse } from "#/src/lib/types/misc";
+import { toResponse } from "#/src/lib/utils";
+import { statusConst } from "#/src/lib/utils/status";
+import { Request as ExpressRequest } from "express"; // Explicitly import from Express
+import {
+  Controller,
+  Delete,
+  FormField,
+  Get,
+  Patch,
+  Request,
+  Route,
+  Security,
+  Tags,
+  UploadedFile,
+} from "tsoa";
+import { getReqUser } from "../auth/auth.helpers";
+
+import fileService from "../file/file.service";
+import { sanitizeUser } from "../user/user.helpers";
+import userService from "../user/user.service";
+import { SanitizedUser } from "../user/user.types";
+import { profileService } from "./profile.services";
+@Route("profile")
+@Tags("Profile")
+@Security("jwt")
+export class ProfileController extends Controller {
+  @Get("/")
+  public async fetch(
+    @Request() req: ExpressRequest
+  ): Promise<APIResponse<SanitizedUser>> {
+    const reqUser = getReqUser(req);
+
+    const user = await userService.fetch(reqUser.id);
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
+
+    return toResponse({ data: sanitizeUser(user) });
+  }
+  @Patch("/")
+  public async update(
+    @Request() req: ExpressRequest,
+    @FormField() name?: string,
+    @FormField() bio?: string,
+    @FormField() add_fcm_token?: string,
+    @FormField() remove_fcm_token?: string,
+    @UploadedFile() photo?: Express.Multer.File
+  ): Promise<APIResponse<SanitizedUser>> {
+    const reqUser = getReqUser(req);
+    if (add_fcm_token && remove_fcm_token) {
+      this.setStatus(statusConst.invalidData.code);
+      return toResponse({ error: "can not add and remove token at once" });
+    }
+    if (add_fcm_token) {
+      await profileService.addFcmToken(reqUser.id, add_fcm_token);
+    }
+    if (remove_fcm_token) {
+      await profileService.removeFCMToken(reqUser.id, remove_fcm_token);
+    }
+
+    let photoUrl = "";
+    if (photo) {
+      [photoUrl] = await fileService.save([{ ...photo }]);
+    }
+
+    const data = {
+      name,
+      bio,
+      photo: photoUrl,
+    };
+
+    const filteredData = Object.fromEntries(
+      Object.entries(data).filter(
+        ([_, v]) => v !== undefined && v !== null && v !== ""
+      )
+    );
+
+    const user = await userService.update(reqUser.id, filteredData);
```

Comment:
We can potentially make multiple calls to the db if any other field is present in addition to the added or removed token.

Let's optimize this and create a single profile service to cater the updated and the token addition / removal at the same time.

The desired way would look something like:
```suggestion
    const data = {
      name,
      bio,
      photo: photoUrl,
      added_fcm_token,
      removed_fcm_token
    };

    const filteredData = Object.fromEntries(
      Object.entries(data).filter(
        ([_, v]) => v !== undefined && v !== null && v !== ""
      )
    );

    const user = await profileService.update(reqUser.id, filteredData);
```